demo/lightning/checkpoint/multinode/train.py

```diff
@@ -0,0 +1,237 @@
+import io
+import os
+import socket
+import time
+import torch
+
+from contextlib import contextmanager
+from typing import Generator
+from lightning import Trainer
+from lightning.pytorch.callbacks import ModelCheckpoint
+from lightning.pytorch.demos import (LightningTransformer, Transformer,
+                                     WikiText2)
+from torch.utils.data import DataLoader
+from pathlib import Path
+from google.cloud import storage
+
+from dataflux_pytorch.lightning.gcs_filesystem import GCSDistributedWriter, GCSDistributedReader
+from dataflux_pytorch.lightning.path_utils import parse_gcs_path
+from dataflux_core import user_agent
+from dataflux_pytorch.lightning import DatafluxLightningCheckpoint
+from lightning.pytorch.trainer.states import TrainerFn
+from lightning.pytorch.strategies import FSDPStrategy
+from lightning.pytorch.strategies.fsdp import _METADATA_FILENAME
+from torch.distributed.checkpoint import save, load
+from torch.nn import Module
+
+
+class DatafluxFSDPStrategy(FSDPStrategy):
+
+    def __init__(self, path, project_name, storage_client, model, **kwargs):
+        super().__init__(**kwargs)
+        self.writer = GCSDistributedWriter(path, project_name, storage_client)
+        self.reader = GCSDistributedReader(
+            path, project_name, storage_client)
+        self.checkpoint_io = DatafluxLightningCheckpoint(
+            project_name, storage_client)
+        self.model = model
+        self.storage_client = storage.Client(project=project_name,)
```

Comment:
nit: remove the comma after project_name if its the last kwarg

Reply:
Done

Reply:
Don't see this addressed

Reply:
I was yet to push the change. Done